common/test/java/com/couchbase/lite/LiveQueryTest.java

```diff
@@ -51,108 +54,107 @@ public void testCreateBasicListener() throws InterruptedException {
-            assertFalse(latch1.await(2 * TOLERABLE_DELAY_MS, TimeUnit.MILLISECONDS));
+            // creation of the second listener should not trigger first listener callback
+            assertFalse(latch1[1].await(TOLERABLE_QUERY_RUN_TIME * 2, TimeUnit.MILLISECONDS));
 
             createDocNumbered(11);
 
-            // introducing change in database should trigger both listener callbacks
-            assertTrue(latch1.await(TOLERABLE_DELAY_MS, TimeUnit.MILLISECONDS));
-            assertTrue(latch2[1].await(TOLERABLE_DELAY_MS, TimeUnit.MILLISECONDS));
+            // introducing change in database should trigger both listener callbacks after the time it takes
+            // for core to wait and rerun query
+            assertTrue(latch1[1].await(TOLERABLE_DATABASE_CHANGE_DELAY, TimeUnit.MILLISECONDS));
+            assertTrue(latch2[1].await(TOLERABLE_DATABASE_CHANGE_DELAY, TimeUnit.MILLISECONDS));
         }
         finally {
             query.removeChangeListener(token1);
             query.removeChangeListener(token2);
         }
     }
 
-    // Creating a document that a query can see should cause an update
-    @Ignore("Test is flaky with 220ms delay")
-    @Test
-    public void testAddChangeToBasicLiveQuery() throws CouchbaseLiteException, InterruptedException {
-        final Query query = QueryBuilder
-            .select(SelectResult.expression(Meta.id))
-            .from(DataSource.database(baseTestDb))
-            .where(Expression.property(KEY).greaterThanOrEqualTo(Expression.intValue(0)))
-            .orderBy(Ordering.property(KEY).ascending());
-
-        final CountDownLatch latch = new CountDownLatch(1);
-
-        ListenerToken token = query.addChangeListener(testSerialExecutor, change -> latch.countDown());
-
-        createDocNumbered(10);
-
-        try { assertTrue(latch.await(TOLERABLE_DELAY_MS, TimeUnit.MILLISECONDS)); }
-        finally { query.removeChangeListener(token); }
-    }
-
-    // !!! What the ever-luvvin heck is this trying to test???
-    @Ignore("Test fail with 220ms delay")
+    // When a result set is closed, we should still be able to introduce a change
     @Test
     public void testCloseResultsInLiveQueryListener() throws CouchbaseLiteException, InterruptedException {
         final Query query = QueryBuilder
             .select(SelectResult.expression(Meta.id))
             .from(DataSource.database(baseTestDb));
 
         final int[] count = new int[] {0};
+
         final CountDownLatch[] latches = new CountDownLatch[2];
         for (int i = 0; i < latches.length; i++) { latches[i] = new CountDownLatch(1); }
         ListenerToken token = query.addChangeListener(
             testSerialExecutor,
             change -> {
-                // this will close the resultset
                 try (ResultSet rs = change.getResults()) {
-                    int n = count[0]++;
-                    latches[n].countDown();
+                    if (rs != null) {
```

Comment:
This is really weird.  The try-with-resources is going to close the rs, but it is closed again, in the `if`.  What is this supposed to be doing?

I probably failed to notice that, last time I looked at this....  Sorry.